### PR TITLE
UI Bug Render Overflow in Bottom Nav Bar.

### PR DESCRIPTION
**Pull request by AbdullahAhmedSoomro**
- Refactor bottom nav bar.

### CODE REVIEW

File: lib/presentation/screen/main_screen.dart [hunk 1]
```diff
@@ -40,45 +40,34 @@ class _MainScreenState extends State<MainScreen> {
...
+      ),
+      bottomNavigationBar: CustomBottomNavBar(
+        backgroundColor: Colors.white,
+        selectedItemColor: const Color(0xFF7B7CFF),
+        unselectedItemColor: const Color(0xFF7B7CFF).withOpacity(0.4),
+        selectedLabelStyle: Theme.of(context).textTheme.bodyMedium!.copyWith(
+            fontWeight: FontWeight.w500, color: const Color(0xFF7B7CFF)),
+        unselectedLabelStyle: Theme.of(context).textTheme.bodyMedium!.copyWith(
+            fontWeight: FontWeight.w400,
+            color: const Color(0xFF7B7CFF).withOpacity(0.4)),
```
Comment (hashirshoaeb) [inline lib/presentation/screen/main_screen.dart:56]: define the style inside, no need to pass from outside

File: lib/presentation/screen/main_screen.dart [hunk 2]
```diff
@@ -40,45 +40,34 @@ class _MainScreenState extends State<MainScreen> {
   @override
   Widget build(BuildContext context) {
     return GradientScaffold(
```
Comment (hashirshoaeb) [inline lib/presentation/screen/main_screen.dart:42]: Remove Gradient scaffold from the tabs widget, (Home Screen and Profile Screen). simply use transparent scaffold for appbar.

File: lib/presentation/shared/custom_bottom_nav_bar.dart [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import 'package:flutter/material.dart';
 import 'package:star_book/presentation/theme/styling/theme_color_style.dart';
 
-class CustomBottomNavBar extends StatefulWidget {
+class BottomNavBar extends StatefulWidget {
```
Comment (hashirshoaeb) [inline lib/presentation/shared/custom_bottom_nav_bar.dart:4]: why stateful when it has no state?

Comment (AbdullahAhmedSoomro) [reply lib/presentation/shared/custom_bottom_nav_bar.dart]: Refactored

**Review by hashirshoaeb — APPROVED** [review]

**Pull request merged**